### DISCUSSION TITLE
[FEATURE] Support a "metadata" kvssink param

### DISCUSSION THREAD

**Discussion by richardgavel-ordinaryexperts** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1031] [Ideas]
**Is your feature request related to a problem? Please describe.**
Unable to add fragment metadata when using the gst-launch-1.0 command line.

**Describe the solution you'd like**
A kvssink param similar to the "stream_tags" param, but instead of being used for stream creation, it's a set of name/value pairs to use to invoke the putFragmentMetadata method after starting the stream.

**Comment by hassanctech** [comment]
https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/pull/1015

Considering this ^.

Also, the PutFragmentMetadata API allows for a new/different tag per fragment.  I'm considering in addition to this a custom event which client GStreamer applications can put on the event bus which `kvssink` will respond to and invoke the PutFragmentMetadata API.  

However as @richardgavel-ordinaryexperts has suggested, such an approach will not easily work for clients using gst-launch so we will likely need both changes, I can work on the latter.
